src/simulation/Simulation.cpp

```diff
@@ -4897,18 +4888,18 @@ void Simulation::UpdateParticles(int start, int end)
 
 int Simulation::GetParticleType(ByteString type)
 {
-	char * txt = (char*)type.c_str();
-
 	// alternative names for some elements
-	if (!strcasecmp(txt, "C4"))
+	auto txt = type.ToUpper();
+	if (txt == "C4")
 		return PT_PLEX;
-	else if (!strcasecmp(txt, "C5"))
+	else if (txt == "C5")
 		return PT_C5;
-	else if (!strcasecmp(txt, "NONE"))
+	else if (txt == "NONE")
 		return PT_NONE;
-	for (int i = 1; i < PT_NUM; i++)
+
+	for (size_t i = 1; i < PT_NUM; i++)
 	{
-		if (!strcasecmp(txt, elements[i].Name.c_str()) && elements[i].Name.size() && elements[i].Enabled)
+		if (!(type == elements[i].Name) && elements[i].Name.size() && elements[i].Enabled)
```

Comment:
`txt`?

Reply:
I used the ByteString directly as they had comparison operators, i thought i had used the recent addition of case conversion to ensure the names where in the same case, but then just comparison operators rather than strcasecompare so it is easier to read.